RockPaperScissors/RockPaperScissors/main.swift

```diff
@@ -1,10 +1,105 @@
 //
 //  RockPaperScissors - main.swift
-//  Created by yagom. 
+//  Created by Harry, Andrew. 
 //  Copyright © yagom academy. All rights reserved.
 // 
 
 import Foundation
 
-print("Hello, World!")
+func printMenu() {
+    print("가위(1), 바위(2), 보(3)! <종료: 0>", terminator: ": ")
+}
 
+func readUserInput() -> String {
+    guard let userInput = readLine() else {
+        return ""
+    }
+    
+    return userInput
+}
+
+func validationUserInput(userInput: String) throws -> Int {
+    guard userInput != "" else {
+        throw InputError.invalidInput
+    }
+    guard let number = Int(userInput) else {
+        throw InputError.invalidInput
+    }
+    guard number >= 0, number <= 3 else {
+        throw InputError.invalidInput
+    }
+
+    return number
+}
+
+func generateComputerHand() -> HandShape? {
+    guard let computerHand = HandShape.allCases.randomElement() else {
+        return nil
+    }
+    
+    return computerHand
+}
+
+func generateUserHand(validationResult: Int) -> HandShape? {
+    let handShapeList: [HandShape?] = HandShape.allCases
+    
+    guard let userHand = handShapeList[validationResult - 1]  else {
+        return nil
+    }
+    
+    return userHand
+}
+
+func compareHandShape(computerHand: HandShape?, userHand: HandShape?) -> MatchResult? {
+    guard let computerHand = computerHand, let userHand = userHand else {
+        return nil
+    }
+    
+    switch (computerHand, userHand) {
+    case (HandShape.rock, HandShape.scissors),
+        (HandShape.scissors, HandShape.paper),
+        (HandShape.paper, HandShape.rock):
+        return .lose
+    case (HandShape.scissors, HandShape.rock),
+        (HandShape.paper, HandShape.scissors),
+        (HandShape.rock, HandShape.paper):
+        return .win
+    default:
+        return .draw
+    }
+}
+
+func startGame() {
+    var selectGameEnd: Bool = false
+    
+    while !selectGameEnd {
+        printMenu()
+        
+        let input = readUserInput()
+        do {
+            let userNumber = try validationUserInput(userInput: input)
+            if userNumber == 0 {
+                print("게임 종료")
+                selectGameEnd.toggle()
+                continue
+            }
+            let userHand = generateUserHand(validationResult: userNumber)
+            let computerHand = generateComputerHand()
+            
+            if let matchResult = compareHandShape(computerHand: computerHand, userHand: userHand) {
+                switch matchResult {
+                case .win:
+                    print("이겼습니다!")
+                case .draw:
+                    print("비겼습니다!")
+                case .lose:
+                    print("졌습니다!")
+                }
+            }
+        }catch {
+            print("잘못된 입력입니다. 다시 시도해주세요.")
+        }
```

Comment:
```suggestion
        } catch {
            print("잘못된 입력입니다. 다시 시도해주세요.")
        }
```

이 부분은 컨벤션이 맞지 않는 거 같네요. 리팩토링을 할 때 컨벤션을 지켰는지의 여부도 체크하는 습관 중요합니다!

Reply:
넵, 이번엔 저희가 확인이 좀 부족했네요 😂